tiledb/sm/rest/rest_client.cc

```diff
@@ -114,6 +114,80 @@ Status RestClient::set_header(
   return Status::Ok();
 }
 
+tuple<Status, std::optional<bool>> RestClient::check_array_exists_from_rest(
+    const URI& uri) {
+  // Init curl and form the URL
+  Curl curlc;
+  std::string array_ns, array_uri;
+  RETURN_NOT_OK_TUPLE(uri.get_rest_components(&array_ns, &array_uri), nullopt);
+  const std::string cache_key = array_ns + ":" + array_uri;
+  RETURN_NOT_OK_TUPLE(
+      curlc.init(config_, extra_headers_, &redirect_meta_, &redirect_mtx_),
+      nullopt);
+  const std::string url = redirect_uri(cache_key) + "/v1/arrays/" + array_ns +
```

Comment:
No immediate action item, but noticing here that we should eventually encapsulate this URI construction rather than concat'ing strings, in order to facilitate graceful version evolution.

Reply:
Object libraries part 8 #2956 starts down this path by putting URI parsing code (extracted from misc.h) into a new directory `storage_format`. That's the place to put URI building functions as requested.